invenio_rdm_records/services/pids/providers/datacite.py

```diff
@@ -130,7 +129,7 @@ def register(self, pid, record, **kwargs):
             # PIDS-FIXME: move to async task, exception handling included
             try:
                 doc = DataCite43JSONSerializer().dump_one(record)
-                url = kwargs["url"]
+                url = record.to_dict()["links"]["self_html"]
```

Comment:
sanity check: can we do `record.links` or `record["links"]. At this stage is a `RDMRecord` no? and thats why we need `to_dict()`?

Reply:
After discussion, going back to `kwargs["url"]`.